chat-commands.js

```diff
@@ -2479,6 +2479,62 @@ exports.commands = {
 	},
 	blacklisthelp: [`/blacklist [username], [reason] - Blacklists the user from the room you are in for a year. Requires: # & ~`],
 
+	bb: 'battleban',
+	battleban: function (target, room, user) {
+		if (!target) return this.parse('/help battleban');
+
+		target = this.splitTarget(target);
+		let targetUser = this.targetUser;
+		if (!targetUser) return this.errorReply(`User ${this.targetUsername} not found.`);
+		if (target.length > MAX_REASON_LENGTH) {
+			return this.errorReply(`The reason is too long. It cannot exceed ${MAX_REASON_LENGTH} characters.`);
+		}
+		if (!target) {
+			return this.errorReply("Battle banning require a reason.");
+		}
+		if (!this.can('lock', targetUser)) return false;
+		if (targetUser.battlebanned) return this.errorReply(`User '${targetUser.name}' is already banned from battling.`);
+
+		const reasonText = target ? ` (${target})` : `.`;
+		const battlebanMessage = `${targetUser.name} was banned from starting new battles by ${user.name}${reasonText}`;
+		this.privateModAction(`(${battlebanMessage})`);
+
+		// Notify staff room when a user is banned from battling outside of it.
+		if (room.id !== 'staff' && Rooms('staff')) {
+			Rooms('staff').addByUser(user, "<<" + room.id + ">> " + battlebanMessage);
+		}
+
+		this.globalModlog("BATTLEBAN", targetUser, ` by ${user.userid}${reasonText}`);
+		Ladders.cancelSearches(targetUser);
+		Punishments.battleban(targetUser, null, null, target);
+		targetUser.popup(`|modal|${user.name} has prevented you from starting new battles${reasonText}`);
```

Comment:
I still think this should say how long its for

Reply:
Oh sorry I forgot about that